api/api_ethereum_test.go

```diff
@@ -2431,3 +2435,160 @@ func TestEthereumAPI_GetRawTransactionByBlockNumberAndIndex(t *testing.T) {
 
 	mockCtrl.Finish()
 }
+
+type testChainContext struct {
+	header *types.Header
+}
+
+func (mc *testChainContext) Engine() consensus.Engine {
+	return gxhash.NewFaker()
+}
+
+func (mc *testChainContext) GetHeader(common.Hash, uint64) *types.Header {
+	return mc.header
+}
+
+// revert("hello")
+//   PUSH1 73; PUSH1 12; PUSH1 0; CODECOPY // mem[0:0+73] = code[12:12+73]
+//   PUSH1 73; PUSH1 0; REVERT // revert mem[0:0+73]
+var codeRevertHello = "0x6049600c60003960496000fd08c379a00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000568656c6c6f"
```

Comment:
Q1. I've hand-crafted it with the help of https://www.evm.codes. It has the [playground mode]( https://www.evm.codes/playground) where you can debug assembly. 
Q2. Because when I compiled one in Remix the bytecode was quite long (didn't exactly remember the code). But yours looks clearer and short enough. Fixed in e7ce4fd1.

Reply:
Q1. Thank you!

> Now I am curious how constructor() gets invoked after deployment..


In my best understanding, let me try to explain it . If you notice any inaccuracies, I welcome your guidance.

When following the regular path, like using a `Create()` function, the `constructor()` code will not exist in the database since it's executed only once during contract creation and its code is not stored. However, in the test contract code, we have hardcoded the `constructor()` function, which is not something that can happen in a real-world scenario. Consequently, the `constructor()` function remains present and executable, even though it should have been removed in a practical deployment.

Reply:
I see. Can you show me the source code and the `solc` command line and version? I'm not reproducing the result right now.

Reply:
You can reproduce in the Remix with the solidity version 0.8.18 and optimization runs value `200`.